.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+# Checklist
+
+- Branch
+    - [ ] Commit sequence broadly makes sense
+    - [ ] Commits have useful messages
+    - [ ] At every intermediate commit, the libraries must compile. It is fine if tests don't compile/pass at these commits
```

Comment:
We haven't been following that.  Instead, we often refactored one library and in the following commit updated all the dependencies. This makes it much nicer for a review, though slightly less convenient when bisecting trying to find which change introduced a bug.

Reply:
Removed :+1: